lang/src/org/partiql/lang/typesystem/builtin/types/nonprimitive/CharType.kt

```diff
@@ -0,0 +1,37 @@
+package org.partiql.lang.typesystem.builtin.types.nonprimitive
+
+import org.partiql.lang.eval.ExprValue
+import org.partiql.lang.eval.ExprValueType
+import org.partiql.lang.typesystem.builtin.types.primitive.IntType
+import org.partiql.lang.typesystem.builtin.types.valueFactory
+import org.partiql.lang.typesystem.interfaces.type.BuiltInType
+import org.partiql.lang.typesystem.interfaces.type.ParametricType
+import org.partiql.lang.typesystem.interfaces.type.SqlType
+import org.partiql.lang.typesystem.interfaces.type.TypeParameters
+import org.partiql.lang.util.asIonInt
+
+object CharType : BuiltInType(), ParametricType {
+    override val typeAliases: List<String>
+        get() = listOf("char", "character")
+
+    override val exprValueType: ExprValueType
+        get() = ExprValueType.STRING
+
+    override val isPrimitiveType: Boolean
+        get() = false
+
+    override val requiredParameters: List<SqlType> = emptyList()
+
+    override val optionalParameters: List<Pair<SqlType, ExprValue>> = listOf(
+        IntType to valueFactory.newInt(Integer.MAX_VALUE)
```

Comment:
Curious, why `MAX_VALUE`?

Reply:
I think trinodb does the same, https://github.com/trinodb/trino/blob/master/core/trino-spi/src/main/java/io/trino/spi/type/VarcharType.java#L47

Reply:
But do we know why?

Reply:
This is just my speculation. I think normally, the parameter of `VARCHAR` is not going to be too large, so `Integer.MAX_VALUE` is large enough to deal with all the use cases.

Reply:
Probably, we can also add a constraint in `validateParameters` method and requires the parameter to be less than `Integer.MAX_VALUE`.

Reply:
Sorry for the confusion. Actually, TrinoDB uses `Integer.MAX_VALUE` to represent unbounded length, and requires the parameter of `VARCHAR` to be less than `Integer.MAX_VALUE`.